src/rules/unnecessary_function_arguments.erl

```diff
@@ -3,24 +3,111 @@
 %%      ignored in all function clauses.</p>
 %%      <p>To avoid this warning, remove the unused argument(s).</p>
 %%      <p><b>Note:</b> This rule will not emit a warning if the function
-%%      implements a behaviour callback or a NIF call.</p>
+%%      implements a NIF call.</p>
 -module(unnecessary_function_arguments).
 
 -behaviour(hank_rule).
 
 -export([analyze/2, ignored/2]).
 
+%% To allow erl_syntax:syntaxTree/0 type spec
+-elvis([{elvis_style, atom_naming_convention, #{regex => "^([a-zA-Z][a-z0-9]*_?)*$"}},
+        {elvis_style, invalid_dynamic_call, disable},
+        {elvis_style, nesting_level, [6]}]).
+
+-type imp_callbacks() :: #{File :: string() => [tuple()] | ignore}.
+
 %% @private
 -spec analyze(hank_rule:asts(), hank_context:t()) -> [hank_rule:result()].
 analyze(FilesAndASTs, _Context) ->
+    ImpCallbacks = callback_usage(FilesAndASTs),
     [Result
      || {File, AST} <- FilesAndASTs,
-        not hank_utils:implements_behaviour(AST),
         not hank_utils:is_old_test_suite(File),
+        not is_unrecognized_behaviour(File, ImpCallbacks),
         Node <- AST,
         erl_syntax:type(Node) == function,
+        is_not_a_callback(ImpCallbacks, File, Node),
```

Comment:
Consider flipping it…
```suggestion
        not is_callback(ImpCallbacks, File, Node),
```

Reply:
I would also put the `Node` parameter as the first one, since that's how we usually organize functions here, i.e. `function_name(DirectObject, OtherContext, MoreContext, Etc)`